src/transaction.cc

```diff
@@ -85,13 +85,13 @@ void TransactionRuleMessageManagement::logMatchLastRuleOnTheChain(const RuleWith
 }
 
 void TransactionRuleMessageManagement::messageNew() {
-    m_rulesMessages.push_back(new RuleMessage(m_transaction));
+    m_rulesMessages.push_back(std::shared_ptr<RuleMessage>( new RuleMessage(m_transaction)));
```

Comment:
Maybe we should eventually use `std::make_shared<T>(...)` everywhere? It uses a single allocation for both reference-counting control block and the data, whereas `std::shared_ptr<T>(new T(...))` uses a separate allocation for the control block.